src/house.h

```diff
@@ -234,7 +235,7 @@ class House
 		Container transfer_container{ITEM_LOCKER1};
 
 		HouseTileList houseTiles;
-		std::list<Door*> doorList;
+		std::set<Door*> doorSet;
```

Comment:
I'm not sure the overhead is worth it. Most houses don't even have internal doors, and guild halls generally between 4 and 8 AFAIK.